wagtail/core/models.py

```diff
@@ -106,29 +94,41 @@ def _make_copy(source, data_dict, m2m_dict, update_attrs=None):
 
     if update_attrs:
         for field, value in update_attrs.items():
-            if field in m2m_dict:
-                continue
```

Comment:
As far as I can see, this check hasn't been accounted for in the new code. If `update_attrs` contains any (non-parental) M2M fields, then the following `setattr` will fail because `target` is unsaved at this point.

Reply:
Replacing this with

    if field not in data_dict:
        continue

should work, I think, since `data_dict` should contain exactly the fields that are legal to update at this point. (Unless there's a discrepancy between `foo` and `foo_id`, possibly...)

Reply:
actually, that will cause any unrecognised items in `update_attrs` to be silently discarded rather than raising an exception, which probably isn't ideal :-/